src/os/alert/alertmanager.js

```diff
@@ -1,163 +1,161 @@
-goog.provide('os.alert.AlertManager');
-goog.provide('os.alertManager');
+goog.module('os.alert.AlertManager');
+goog.module.declareLegacyNamespace();
+
 goog.require('goog.events.EventTarget');
 goog.require('goog.log');
 goog.require('goog.structs.CircularBuffer');
 goog.require('os.alert.AlertEvent');
-goog.require('os.alert.AlertEventSeverity');
 goog.require('os.array');
 goog.require('os.structs.EventType');
 
+const AlertEventSeverity = goog.require('os.alert.AlertEventSeverity');
+const AlertEventType = goog.require('os.alert.EventType');
+
 
+/**
+ * Maximum number of alerts to be saved.
+ * @type {number}
+ * @private
+ */
+const maxSaved_ = 10000;
 
 /**
  * Responsible for receiving, logging and reporting alerts
- *
  * @extends {goog.events.EventTarget}
- * @constructor
  */
-os.alert.AlertManager = function() {
-  os.alert.AlertManager.base(this, 'constructor');
-
+class AlertManager extends goog.events.EventTarget {
   /**
-   * @type {!goog.structs.CircularBuffer.<!os.alert.AlertEvent>}
-   * @private
    */
-  this.savedAlerts_ = new goog.structs.CircularBuffer(os.alert.AlertManager.MAX_SAVED);
+  constructor() {
+    super();
+
+    /**
+     * @type {!goog.structs.CircularBuffer.<!os.alert.AlertEvent>}
+     * @private
+     */
+    this.savedAlerts_ = new goog.structs.CircularBuffer(maxSaved_);
+
+    /**
+     * Tracks the number of alerts process by any given client so the client can get all alerts even if it comes online
+     * later.
+     * @type {!Object.<!string, !number>}
+     * @private
+     */
+    this.processedByClientCount_ = {};
+
+    /**
+     * Ensures that processMissedAlerts is only called once
+     * @type {boolean}
+     * @private
+     */
+    this.missedAlertsProcessed_ = false;
+
+    /**
+     * Ensures that an alert is only sent once
+     * @type {Object<string, boolean>}
+     * @private
+     */
+    this.onceMap_ = {};
+  }
 
   /**
-   * Tracks the number of alerts process by any given client so the client can get all alerts even if it comes online
-   * later.
-   * @type {!Object.<!string, !number>}
-   * @private
+   * Maximum number of alerts to be saved.
+   * @return {number}
    */
-  this.processedByClientCount_ = {};
+  static get MAX_SAVED() {
```

Comment:
Do we have an example of declaring a setter?

Reply:
Yes: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Functions/set

Note this is using `static get` to create a constant property on the class itself (`os.alert.AlertManager.MAX_SAVED`). This seems to be the [recommended way](https://stackoverflow.com/a/32647583) to define a constant on an ES6 class.